Corona-Warn-App/src/main/java/de/rki/coronawarnapp/appconfig/ConfigChangeDetector.kt

```diff
@@ -60,4 +64,19 @@ class ConfigChangeDetector @Inject constructor(
             RiskLevelRepository.setRiskLevelScore(RiskLevel.UNDETERMINED)
         }
     }
+
+    @VisibleForTesting
+    internal suspend fun checkForDiagnosisKeysDataMapping(exposureWindowRiskCalculationConfig: ExposureWindowRiskCalculationConfig) {
+        val oldDiagnosisKeyDataMapping = enfClient.getDiagnosisKeysDataMapping()
+        val newDiagnosisKeyDataMapping = exposureWindowRiskCalculationConfig.diagnosisKeyDataMapping
+
+        if (oldDiagnosisKeyDataMapping != newDiagnosisKeyDataMapping) {
+            try {
+                Timber.i("New DiagnosisKeysDataMapping differs from last one, applying.")
+                enfClient.setDiagnosisKeysDataMapping(newDiagnosisKeyDataMapping)
```

Comment:
This can throw exceptions for various reasons, at least:
1. rate limit error
1. invalid param

Only the rate limit error should be silent, other errors should not (otherwise, we wouldn't figure out that there's an error)

Reply:
new implementation rethrows error if status code is not Rate Limit